src/Features/Core/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -355,7 +366,14 @@ private bool CheckOption(Workspace workspace, string language, bool documentOpen
                     continue;
```

Comment:
We do a `state.Remove(project.Id)` at the start of method, but persist only document diagnostics data here, not project diagnostics with no location? Where are those persisted?